### PR TITLE
Improve plot_state_histogram configuration options.

### PR DESCRIPTION
**Pull request by tanujkhattar**
This PR introduces the following changes
- Make `tick_labels` (and other plot properties) configurable so that users can provide different labels
- Changes default tick labels from binary string ('0111') to integers ('7') to make larger plots more readable. 
- Add support for `collections.Counter` so users can directly pass the result of `result.histogram()` so that processed results can be plotted easily. 

Once this is checked-in, I'll send a PR for state_histogram.ipynb tutorial.

### CODE REVIEW

File: cirq/vis/state_histogram.py [hunk 1]
```diff
@@ -50,32 +50,54 @@ def get_state_histogram(result: 'result.Result') -> np.ndarray:
...
 
 def plot_state_histogram(
-    values: Union['result.Result', np.ndarray], ax: Optional['plt.Axis'] = None
+    data: Union['result.Result', collections.Counter, Sequence[SupportsFloat]],
+    ax: Optional['plt.Axis'] = None,
+    *,
+    tick_label: Optional[Sequence[str]] = None,
+    xlabel: Optional[str] = 'qubit state',
+    ylabel: Optional[str] = 'result count',
+    title: Optional[str] = 'Result State Histogram',
```
Comment (95-martin-orion) [inline cirq/vis/state_histogram.py:59]: Do we want to allow `None` for these fields? If not, these shouldn't have type `Optional`.

**Review by 95-martin-orion — CHANGES_REQUESTED** [review]
Mostly LGTM, just a couple details to clear up.

File: cirq/vis/state_histogram.py [hunk 2]
```diff
@@ -50,32 +50,54 @@ def get_state_histogram(result: 'result.Result') -> np.ndarray:
...
-        ax:     The Axes to plot on. If not given, a new figure is created,
-                plotted on, and shown.
+        data:   The histogram values to plot. Possible options are:
+                `result.Result`: Histogram is computed using
+                `get_state_histogram` and all 2 ** num_qubits values are
+                 plotted, including 0s.
+                `collections.Counter`: Only (key, value) pairs present in
+                 collection are plotted.
+                `Sequence[SupportsFloat]`: Values in input are plotted with
+                 default labels as |0> ... |n-1>.
```
[the remark this answers is not in the document]
Comment (tanujkhattar) [reply cirq/vis/state_histogram.py]: done.

File: cirq/vis/state_histogram.py (hunk 1 above)
Comment (tanujkhattar) [reply cirq/vis/state_histogram.py:59]: Yes, these fields can potentially be none. For eg, we might want to omit the title when drawing a figure to be included in a pdf. 
We also follow the same pattern for other visualization utilities, for eg: `integrated_histogram` 
https://github.com/quantumlib/Cirq/blob/bee1a448bc98b52ab952ccf284c0382918abaad1/cirq/vis/histogram.py#L32

**Review by 95-martin-orion — APPROVED** [review]

**Pull request merged**